scripts/codepregen.py

```diff
@@ -74,7 +74,7 @@ def _ParallelGenerateOne(arg):
 @click.option(
     '--generator',
     default='all',
-    type=click.Choice(['all', 'zap', 'codegen']),
+    type=click.Choice(['all', 'zap', 'codegen', 'codegen-cpp-only']),
```

Comment:
why is this a separate generator? We generally have two code generators: zap and python (codegen).

codegen-cpp-only seems odd ... should this be some sort of flag instead? For some purposes at least, codegen-cpp-only seems to behave exactly like codegen. Is this only used to filter out java? and if so, why do we care?

Reply:
> Is this only used to filter out java?

Yes.

> and if so, why do we care?

Because having the java stuff checked in under `zzz_generated/darwin` is just confusing noise.